source/change-streams/tests/unified/change-streams.yml

```diff
@@ -0,0 +1,74 @@
+description: "change-streams"
+schemaVersion: "1.0"
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: []
+      ignoreCommandMonitoringEvents: []
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: *database0
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: *collection0
+initialData:
+  - collectionName: *collection0
+    databaseName: *database0
+    documents: []
+tests:
+  - description: "Test array truncation"
+    runOnRequirements:
+      - minServerVersion: "4.7"
+        topologies: [replicaset]
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: {
+            "_id": 1,
+            "a": 1,
+            "array": ["foo", {"a": "bar"}, 1, 2, 3]
+          }
+      - name: createChangeStream
+        object: *collection0
+        saveResultAsEntity: &changeStream0 changeStream0
+      - name: updateOne
+        object: *collection0
+        arguments:
+          filter: {
+            "_id": 1
+          }
+          update: [
+            {
+              "$set": {
+                "array": ["foo", {"a": "bar"}]
+              }
+            }
+          ]
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult: {
+          "operationType": "update",
+          "ns": {
+            "db": "database0",
+            "coll": "collection0"
+          },
+          # This expectation is based on the MongoDB server behavior
+          # (as opposed to be based on its API),
```

Comment:
The server doesn't have a formal API specification.  Can you elaborate on what you mean by this so we can decide whether this comment makes sense?  This is a public repository so we have to be clear as to the meaning of the terms we use.

Reply:
I think it does. The API<sup>1</sup> is specified in multiple places in the official docs, but mostly by https://docs.mongodb.com/manual/reference/command/. The [design](https://docs.google.com/document/d/1sxGrck8X0r0W37UVLFwV8vaUQv3OM8xlyWWvHB5M2ew/edit#heading=h.22yp33tzmzpp) for the functionality that caused this PR also uses the term "API".

----
<sup>1</sup> I believe, previously people used the term "protocol" for this, and API was used as a name for *programming* interfaces. Now the terms "protocol" and "API" seem to be mixed for no good reason known to me.

Reply:
> what you mean by this

It's up to the server to decide whether to report a change as truncation or not, so even if the change can be represented as truncation, it does not mean it will. How the decision is made is an implementation detail which may change.

Reply:
OK, I see.  I agree with you about the command reference docs being the closest thing we have to an API specification for the server.  So even if we wanted to we couldn't base this test on the API, since the docs for truncatedArrays has yet to be published or written yet.  But once it has it may turn out to be sufficient as the basis for this test. In other words it may be inaccurate to say this is not part of the API, since we don't yet know what the API will specify (see https://jira.mongodb.org/browse/DOCS-14126).

Any other reviewers have thoughts on this?  It's a minor point, and I'm ok to resolve if no one else thinks it's important.

Reply:
I am claiming that judging based on the [design](https://docs.google.com/document/d/1sxGrck8X0r0W37UVLFwV8vaUQv3OM8xlyWWvHB5M2ew/edit#heading=h.c4056nub8lzp) of the server feature, what we relying on in this test is implementation-specific behavior. I also expressed this in the specification change: https://github.com/mongodb/specifications/pull/909/files#diff-b1cdd6bd41c94cc2b4ba9bef6f89e372fcc59b67dc96f349a95a85f9178095d4R218

Reply:
I also agree that command reference docs are the closest to a server API spec in today's world and that the design doc for this feature didn't fully clarify which situations would lead to array changes being reported in `truncatedArrays` vs `updatedFields` or `removedFields`. However, the design isn't meant to be public documentation and as @jyemin mentioned, the public documentation for this feature hasn't been written yet, so maybe we should avoid saying this in a public comment.

If we feel that this comment is useful, maybe we can change it to "the inserted document and the update pipeline were determined empirically by testing out different combinations until change stream event returned a "truncatedArrays" field"? This isn't necessarily the exact wording, but it avoids the possibly-contentious mentioning of "server API".

Reply:
Done.